### PR TITLE
Oauth fe

### PR DESCRIPTION
**Pull request by IamNorma**
Sets up OAuth and sessions

- Adds omiauth and google-client gems
- Creates dashboard and home page
- Adds stub_omniauth in rails_helper for testing

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+class SessionsController < ApplicationController
+  def create
+    conn = Faraday.new(url: "#{ENV['RAILS_ENGINE_DOMAIN']}")
+    response = conn.post('/api/v1/users', auth_hash)
+    json = JSON.parse(response.body, symbolize_names: true)
+    user = User.new(json[:data])
+    session[:user_id] = user.id
+    flash[:notice] = 'You have successfully logged in'
+    redirect_to dashboard_path
+  end
```
Comment (hashmaster3k) [inline app/controllers/sessions_controller.rb:12]: Good start! Will definitely want to refactor into facade, service system later.

**Comment by hashmaster3k** [comment]
Looks great overall! We will want to add a description about the app on the home page as indicated by the wireframes. There were some merge conflicts residing in the POROS and Gemfiles, I have address them. You do have a failing test. I think it MIGHT be related to your key/values in the application.yml. I deleted that file because it was showing up in the pull request. Double check your setup in Travis and Heroku to ensure they have access to those key/values.

**Review by hashmaster3k — CHANGES_REQUESTED** [review]
See notes above.

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -70,6 +70,31 @@
...
+      },
+      credentials: {
+          token: 'MOCK_OMNIAUTH_GOOGLE_TOKEN',
+          refresh_token: 'MOCK_OMNIAUTH_GOOGLE_REFRESH TOKEN',
+          expires_at: DateTime.now,
+          expires: true
+      },
+  }
+  OmniAuth.config.mock_auth[:google_oauth2] = OmniAuth::AuthHash.new(omniauth_google_hash)
+end
```
Comment (LHJE) [inline spec/rails_helper.rb:97]: This is all so good!

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -1,12 +1,23 @@
...
     flash[:success] = "You've made an impact today.  We hope to see you again tomorrow."
     redirect_to root_path
-  end
+  end 
+  
+  private
+
+  def auth_hash
+    request.env["omniauth.auth"]
+  end
```
Comment (LHJE) [inline app/controllers/sessions_controller.rb:22]: All of this looks so great!

**Review by LHJE — APPROVED** [review]
Very impressed Norma!

**Review by leahriffell — APPROVED** [review]
Thanks for your hard work on this!

**Pull request merged**